qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -367,9 +367,73 @@ def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_acti
     return spinz_matrix_elements
 
 
+def get_particle_number_table(mol_name, hf_data, n_active_electrons=None, n_active_orbitals=None):
+    r"""Reads the Hartree-Fock (HF) electronic structure data file, defines an active
+    space and generates the table required to build the particle number operator
+    :math:`\hat{N}=\sum_\alpha \hat{n}_\alpha`.
```

Comment:
The docstring summary emphasizes as well that this function is attempting to do quite a lot 🙂 The docstring summary shouldn't normally require commas/read like a list of bullet points!